src/Helpers/Session/DonationConfirmation/Frontend.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Give\Helper\Session\DonationConfirmation;
+
+use function Give\Helper\Session\getSession;
+use function Give\Helper\Session\removeDataFromSession;
+use function Give\Helper\Session\storeDataIntoSession;
+use function Give\Helpers\Form\Utils\isConfirmingDonation;
+
+/**
+ * Store posted data to donor session to access it in iframe if we are on payment confirmation page.
+ *
+ * Note: only for internal use.
+ *
+ * @return bool
+ * @since 2.7.0
+ */
+function storePostedData() {
+	if ( $_REQUEST['giveDonationAction'] && 'showReceipt' === give_clean( $_REQUEST['giveDonationAction'] ) ) {
+		$paymentGatewayId = ucfirst( give_clean( $_GET['payment-confirmation'] ) );
+		storeDataIntoSession( "postDataFor{$paymentGatewayId}", array_map( 'give_clean', $_POST ) );
+
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * Remove posted data from donor session just before rendering payment confirmation view because beyond this view this data is not useful.
+ *
+ * Note: Only for internal use.
+ *
+ * @return bool
+ * @since 2.7.0
+ */
+function removePostedData() {
+	$paymentGatewayId = ucfirst( give_clean( $_GET['payment-confirmation'] ) );
+	removeDataFromSession( "postDataFor{$paymentGatewayId}" );
+
+	return false;
```

Comment:
Why are you returning false? If you don't need to return then just let it implicitly return void.

Reply:
I do not make sense to`return` any, so removing this.

Reply:
https://github.com/impress-org/givewp/compare/04064e2ec6fd...ddd8d96e937e